docs/code/introduction/machine_code.md

```diff
@@ -0,0 +1,257 @@
+---
+title: "How to add machine-wide custom code"
+---
+
+
+# How to add machine-wide custom code
+
+MPF contains a “CustomCode” concept which lets you add custom code to your game.
+
+CustomCode classes are Python modules that run at the “root” of your game. You can use them to do anything you want.
+
+Note that MPF also has the ability to run custom mode code which is code that is associated with a certain game mode and is generally only active when the mode it’s in is active. So if you just want to write your own custom game logic, you’ll probably use mode code.
+
+CustomCode classes, on the other hand, are sort of “machine-level” custom code. CustomCode classes are nice if you have some kind of custom device type that doesn’t match up to any of MPF’s built in devices. The elevator and claw unloader in Demolition Man is a good example, and what we’ll use here.
+
+(You can read about how to download and run Demo Man in the example games section section of the MPF User Documentation.)
+
+Here’s how to create a custom code class:
+## Create your custom code file
+
+First, add a `custom_code` folder to your machine folder (you can use another name if you want). Then inside there, create the Python file that will hold your custom code classes. You can name this file whatever you want, just remember the name for the next step.
```

Comment:
I mean yes technically you can use another name if you want, but let's not encourage that 😛

Reply:
fair enough. I am not able to get it work with another name anyways. So I removed this encouragement now, came from the original docs.